### ISSUE TITLE
zm10 mobil webmail problem

### ISSUE THREAD

**Issue by linuxgokan** [ianw1974/zimbra-build-scripts#16]
Webmail automatically redirects to "/modern" on mobile phones.

How can I turn this off?

Thanks.

**Comment by ianw1974** [comment]
You have to ask on Zimbra forums about that.  My issues are for the build process.  For usage of Zimbra, that is something else for Zimbra forums.  Go to https://forums.zimbra.com

**Comment by ianw1974** [comment]
Hi, also take a look at this: https://github.com/ianw1974/zimbra-build-scripts/issues/17

Someone else opened an issue, seems there are some changes so I might be able to patch it and release in a new build.

**Comment by ianw1974** [comment]
New builds for ZCS10 are online with mobile url fix that allows for it to be disabled.

**Comment by linuxgokan** [comment]
very thanks.

**Comment by ErickPH** [comment]
@linuxgokan I may be wrong but looking at https://github.com/Zimbra/zm-web-client/pull/822/files
if the code is present on the new build then it looks like this has to be enabled:
`In domain settings

this.registerSetting("DISABLE_MODERN_CLIENT",			{name:"zimbraModernWebClientDisabled", type:ZmSetting.T_DOMAIN, dataType:ZmSetting.D_BOOLEAN, defaultValue:false});`

So there should be a new option there that will allow disabling the modern UI completely.
I will install it tomorrow and test it as today (sunday) I have to do the lawn. :)

**Comment by linuxgokan** [comment]
Yes, line 2048 looks correct to me. As follows... But the result did not change. 
(I did a fresh install today.)

        if (appCtxt.get(ZmSetting.DISABLE_MODERN_CLIENT) !== true) {
            mi = menu.createMenuItem("modernClientLink", {text: ZmMsg.modernClient});
            mi.addSelectionListener(ZmZimbraMail.modernClientLinkCallback);
        }

**Comment by ianw1974** [comment]
Since it's Zimbra code in their repositories, we just have to wait until they fix it.  This is independant of the build process.  The build scripts and build process, builds on what code is available.  Sounds like the patch hasn't been integrated properly.

I don't develop Zimbra, so I cannot help with that.  The build scripts pulls their code and that is pretty much all.

**Comment by vnngoanhtuan** [comment]
Dear All,

We can use configuration of COS to set default WebUI -> Classic and use comment to disable WebUI -> Modern model.

- $zmprov mcf zimbraModernWebClientDisabled TRUE

Thanks,
Tuan Ngo

**Comment by linuxgokan** [comment]
Thank you for the information. But the problem persists on mobile. Fixing COS doesn't solve mobile issue.

**Comment by ianw1974** [comment]
I just edited "default" COS on a fresh Zimbra install and set it to classic.  This worked, both admin user and new users created then defaulted to classic.

I did also run the zmprov command above as well.

**Comment by linuxgokan** [comment]
yes i did that too. The problem is only on mobile. We cannot access webmail from mobile. "modern" redirects.

**Comment by ianw1974** [comment]
Better will be to write about it here: https://forums.zimbra.org/

I can only support what my scripts do.  Anything else is Zimbra reponsibility.  I have already mentioned in some internal channels with Zimbra, but until a lot of people bring it up on the Zimbra forums they might not do anything about it.  Even the webmail on a normal computer defaults to Modern with the new Zimbra 10 and it doesn't even include it so it is crippled until it is configured correctly in COS.  As for mobile, that's obviously also a problem or to use IMAP/POP3 instead.

**Comment by linuxgokan** [comment]
I found the following answers on the forum about this. have you ever tried it?
https://github.com/Zimbra/zm-web-client/commit/35a4a5f70286ed75a7437e9a10e61d306b31f670
https://github.com/Zimbra/zm-web-client/commit/be147e3488045cf9556b30deb8ce100785c7fc9c

**Comment by ianw1974** [comment]
Those are commits made in 2021 so I doubt very much that they are not relevant to the current files because they are over 2.5 years old.  The files in the develop branch are considerably different now.

If Zimbra removed those changes, then they need to put them back in.  It is for them to fix their code.  I am not a java programmer, so I am not going to be attempting to fix their java code.  That is their responsibility.
